app/src/lib/shells/linux.ts

```diff
@@ -1,26 +1,81 @@
 import { spawn } from 'child_process'
+import * as fs from 'fs'
+import { assertNever, fatalError } from '../fatal-error'
 import { IFoundShell } from './found-shell'
 
 export enum Shell {
   Gnome = 'gnome-terminal',
+  Tilix = 'tilix',
 }
 
 export const Default = Shell.Gnome
 
 export function parse(label: string): Shell {
+  if (label === Shell.Gnome) {
+    return Shell.Gnome
+  }
+
+  if (label === Shell.Tilix) {
+    return Shell.Tilix
+  }
+
   return Default
 }
 
+async function getPathIfAvailable(path: string): Promise<string | null> {
+  return new Promise<string | null>(resolve => {
```

Comment:
We have a `pathExists` helper function in [`app/src/lib/file-system.ts`](https://github.com/desktop/desktop/blob/1eb1bbad07e5bded5b307d6abceaa074cd51c35b/app/src/lib/file-system.ts#L103) that you can use in place of this code.

```ts
const exists = await pathExists(path)
if (exists) { 
  resolve(path)
} else {
  resolve(null)
}
```